server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -233,6 +257,36 @@ private RemoteRestoreResult executeRestore(
         return RemoteRestoreResult.build(restoreUUID, restoreInfo, allocationService.reroute(updatedState, "restored from remote store"));
     }
 
+    private ClusterState restoreClusterMetadata(ClusterState restoredClusterState, Metadata clusterMetadata) {
+        Metadata.Builder mdBuilder = new Metadata.Builder();
+        if (clusterMetadata.persistentSettings() != null) {
+            Settings settings = clusterMetadata.persistentSettings();
+            clusterService.getClusterSettings().validateUpdate(settings);
+            mdBuilder.persistentSettings(settings);
+        }
+        if (clusterMetadata.templates() != null) {
+            for (final IndexTemplateMetadata cursor : clusterMetadata.templates().values()) {
+                mdBuilder.put(cursor);
+            }
+        }
+        if (clusterMetadata.customs() != null) {
+            for (final Map.Entry<String, Metadata.Custom> cursor : clusterMetadata.customs().entrySet()) {
+                Metadata.Custom cursorValue = cursor.getValue();
+                if (RepositoriesMetadata.TYPE.equals(cursor.getKey())) {
+                    RepositoriesMetadata repositoriesMetadata = (RepositoriesMetadata) cursorValue;
+                    cursorValue = new RepositoriesMetadata(
+                        repositoriesMetadata.repositories()
+                            .stream()
+                            .filter(repository -> SYSTEM_REPOSITORY_SETTING.get(repository.settings()) == false)
```

Comment:
node join flow will add the system repos, is that correct?

Reply:
yes